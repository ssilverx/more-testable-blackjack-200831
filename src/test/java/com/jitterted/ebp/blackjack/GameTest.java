/*
 * Copyright (C) Fortumo OÜ, 2020
 * All rights reserved. Proprietary and confidential.
 * Unauthorized copying of this file, via any medium is strictly prohibited.
 *
 * All rights reserved. Proprietary and confidential.
 * Unauthorized copying of this file, via any medium is strictly prohibited.
 * Distribution, modification, reproduction, merging, publishing, sub-licensing,
 * sale and/or any other use (in binary form or otherwise) of this software is
 * not permitted, unless Fortumo OÜ has explicitly and in writing permitted such
 * use of the software.
 *
 * THIS SOFTWARE IS PROVIDED BY COPYRIGHT HOLDER ''AS IS'' AND ANY
 * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
 * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
 * DISCLAIMED. IN NO EVENT SHALL Fortumo OÜ BE LIABLE FOR ANY
 * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
 * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
 * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
 * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */

package com.jitterted.ebp.blackjack;

import org.approvaltests.Approvals;
import org.approvaltests.core.Options;
import org.approvaltests.legacycode.Range;
import org.approvaltests.reporters.ClipboardReporter;
import org.junit.jupiter.api.DisplayNameGeneration;
import org.junit.jupiter.api.DisplayNameGenerator;
import org.junit.jupiter.api.Test;

import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.PrintStream;
import java.nio.charset.StandardCharsets;
import java.util.Random;

@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
class GameTest {

    @Test
    void stand() {
        Approvals.verifyAll("blackjack", Range.get(1337, 1350), i -> playTheGame("s", i), new Options(new ClipboardReporter()));
    }

    private String playTheGame(String command, long seed) {
        final UserInput userInput = () -> command;

        try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
            System.setOut(new PrintStream(baos, true, StandardCharsets.UTF_8));

            Game.playTestable(userInput, new Random(seed));

            return baos.toString();
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    @Test
    void hit() {
        Approvals.verify(playTheGame("h", 1337L), new Options(new ClipboardReporter()));
    }
}
