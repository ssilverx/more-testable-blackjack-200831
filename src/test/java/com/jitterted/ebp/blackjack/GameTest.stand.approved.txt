blackjack


[107m[2J[1;1H[32mWelcome to[31m Jitterted's[30m BlackJack
[2J[1;1HDealer has: 
[31m┌─────────┐[1B[11D│9        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        9│[1B[11D└─────────┘
[7A[12C┌─────────┐[1B[11D│░░░░░░░░░│[1B[11D│░ J I T ░│[1B[11D│░ T E R ░│[1B[11D│░ T E D ░│[1B[11D│░░░░░░░░░│[1B[11D└─────────┘
Player has: 
[30m┌─────────┐[1B[11D│6        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        6│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│7        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        7│[1B[11D└─────────┘
 (13)
[H]it or [S]tand?
[2J[1;1HDealer has: 
[31m┌─────────┐[1B[11D│9        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        9│[1B[11D└─────────┘[6A[1C[30m┌─────────┐[1B[11D│6        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        6│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│6        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        6│[1B[11D└─────────┘
 (21)

Player has: 
[30m┌─────────┐[1B[11D│6        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        6│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│7        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        7│[1B[11D└─────────┘
 (13)
You lost to the Dealer. 💸
[m

[107m[2J[1;1H[32mWelcome to[31m Jitterted's[30m BlackJack
[2J[1;1HDealer has: 
[31m┌─────────┐[1B[11D│7        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        7│[1B[11D└─────────┘
[7A[12C┌─────────┐[1B[11D│░░░░░░░░░│[1B[11D│░ J I T ░│[1B[11D│░ T E R ░│[1B[11D│░ T E D ░│[1B[11D│░░░░░░░░░│[1B[11D└─────────┘
Player has: 
[31m┌─────────┐[1B[11D│A        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        A│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│6        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        6│[1B[11D└─────────┘
 (17)
[H]it or [S]tand?
[2J[1;1HDealer has: 
[31m┌─────────┐[1B[11D│7        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        7│[1B[11D└─────────┘[6A[1C[30m┌─────────┐[1B[11D│5        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        5│[1B[11D└─────────┘[6A[1C[30m┌─────────┐[1B[11D│9        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        9│[1B[11D└─────────┘
 (21)

Player has: 
[31m┌─────────┐[1B[11D│A        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        A│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│6        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        6│[1B[11D└─────────┘
 (17)
You lost to the Dealer. 💸
[m

[107m[2J[1;1H[32mWelcome to[31m Jitterted's[30m BlackJack
[2J[1;1HDealer has: 
[31m┌─────────┐[1B[11D│10       │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│       10│[1B[11D└─────────┘
[7A[12C┌─────────┐[1B[11D│░░░░░░░░░│[1B[11D│░ J I T ░│[1B[11D│░ T E R ░│[1B[11D│░ T E D ░│[1B[11D│░░░░░░░░░│[1B[11D└─────────┘
Player has: 
[31m┌─────────┐[1B[11D│9        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        9│[1B[11D└─────────┘[6A[1C[30m┌─────────┐[1B[11D│3        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        3│[1B[11D└─────────┘
 (12)
[H]it or [S]tand?
[2J[1;1HDealer has: 
[31m┌─────────┐[1B[11D│10       │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│       10│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│7        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        7│[1B[11D└─────────┘
 (17)

Player has: 
[31m┌─────────┐[1B[11D│9        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        9│[1B[11D└─────────┘[6A[1C[30m┌─────────┐[1B[11D│3        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        3│[1B[11D└─────────┘
 (12)
You lost to the Dealer. 💸
[m

[107m[2J[1;1H[32mWelcome to[31m Jitterted's[30m BlackJack
[2J[1;1HDealer has: 
[31m┌─────────┐[1B[11D│K        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        K│[1B[11D└─────────┘
[7A[12C┌─────────┐[1B[11D│░░░░░░░░░│[1B[11D│░ J I T ░│[1B[11D│░ T E R ░│[1B[11D│░ T E D ░│[1B[11D│░░░░░░░░░│[1B[11D└─────────┘
Player has: 
[30m┌─────────┐[1B[11D│10       │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│       10│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│10       │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│       10│[1B[11D└─────────┘
 (20)
[H]it or [S]tand?
[2J[1;1HDealer has: 
[31m┌─────────┐[1B[11D│K        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        K│[1B[11D└─────────┘[6A[1C[30m┌─────────┐[1B[11D│7        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        7│[1B[11D└─────────┘
 (17)

Player has: 
[30m┌─────────┐[1B[11D│10       │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│       10│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│10       │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│       10│[1B[11D└─────────┘
 (20)
You beat the Dealer! 💵
[m

[107m[2J[1;1H[32mWelcome to[31m Jitterted's[30m BlackJack
[2J[1;1HDealer has: 
[30m┌─────────┐[1B[11D│10       │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│       10│[1B[11D└─────────┘
[7A[12C┌─────────┐[1B[11D│░░░░░░░░░│[1B[11D│░ J I T ░│[1B[11D│░ T E R ░│[1B[11D│░ T E D ░│[1B[11D│░░░░░░░░░│[1B[11D└─────────┘
Player has: 
[30m┌─────────┐[1B[11D│9        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        9│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│8        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        8│[1B[11D└─────────┘
 (17)
[H]it or [S]tand?
[2J[1;1HDealer has: 
[30m┌─────────┐[1B[11D│10       │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│       10│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│6        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        6│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│A        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        A│[1B[11D└─────────┘
 (17)

Player has: 
[30m┌─────────┐[1B[11D│9        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        9│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│8        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        8│[1B[11D└─────────┘
 (17)
Push: The house wins, you Lose. 💸
[m

[107m[2J[1;1H[32mWelcome to[31m Jitterted's[30m BlackJack
[2J[1;1HDealer has: 
[31m┌─────────┐[1B[11D│5        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        5│[1B[11D└─────────┘
[7A[12C┌─────────┐[1B[11D│░░░░░░░░░│[1B[11D│░ J I T ░│[1B[11D│░ T E R ░│[1B[11D│░ T E D ░│[1B[11D│░░░░░░░░░│[1B[11D└─────────┘
Player has: 
[30m┌─────────┐[1B[11D│2        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        2│[1B[11D└─────────┘[6A[1C[30m┌─────────┐[1B[11D│A        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        A│[1B[11D└─────────┘
 (13)
[H]it or [S]tand?
[2J[1;1HDealer has: 
[31m┌─────────┐[1B[11D│5        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        5│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│K        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        K│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│2        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        2│[1B[11D└─────────┘
 (17)

Player has: 
[30m┌─────────┐[1B[11D│2        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        2│[1B[11D└─────────┘[6A[1C[30m┌─────────┐[1B[11D│A        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        A│[1B[11D└─────────┘
 (13)
You lost to the Dealer. 💸
[m

[107m[2J[1;1H[32mWelcome to[31m Jitterted's[30m BlackJack
[2J[1;1HDealer has: 
[30m┌─────────┐[1B[11D│K        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        K│[1B[11D└─────────┘
[7A[12C┌─────────┐[1B[11D│░░░░░░░░░│[1B[11D│░ J I T ░│[1B[11D│░ T E R ░│[1B[11D│░ T E D ░│[1B[11D│░░░░░░░░░│[1B[11D└─────────┘
Player has: 
[31m┌─────────┐[1B[11D│J        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        J│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│9        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        9│[1B[11D└─────────┘
 (19)
[H]it or [S]tand?
[2J[1;1HDealer has: 
[30m┌─────────┐[1B[11D│K        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        K│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│5        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        5│[1B[11D└─────────┘[6A[1C[30m┌─────────┐[1B[11D│J        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        J│[1B[11D└─────────┘
 (25)

Player has: 
[31m┌─────────┐[1B[11D│J        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        J│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│9        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        9│[1B[11D└─────────┘
 (19)
Dealer went BUST, Player wins! Yay for you!! 💵
[m

[107m[2J[1;1H[32mWelcome to[31m Jitterted's[30m BlackJack
[2J[1;1HDealer has: 
[30m┌─────────┐[1B[11D│8        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        8│[1B[11D└─────────┘
[7A[12C┌─────────┐[1B[11D│░░░░░░░░░│[1B[11D│░ J I T ░│[1B[11D│░ T E R ░│[1B[11D│░ T E D ░│[1B[11D│░░░░░░░░░│[1B[11D└─────────┘
Player has: 
[30m┌─────────┐[1B[11D│9        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        9│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│9        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        9│[1B[11D└─────────┘
 (18)
[H]it or [S]tand?
[2J[1;1HDealer has: 
[30m┌─────────┐[1B[11D│8        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        8│[1B[11D└─────────┘[6A[1C[30m┌─────────┐[1B[11D│7        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        7│[1B[11D└─────────┘[6A[1C[30m┌─────────┐[1B[11D│K        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        K│[1B[11D└─────────┘
 (25)

Player has: 
[30m┌─────────┐[1B[11D│9        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        9│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│9        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        9│[1B[11D└─────────┘
 (18)
Dealer went BUST, Player wins! Yay for you!! 💵
[m

[107m[2J[1;1H[32mWelcome to[31m Jitterted's[30m BlackJack
[2J[1;1HDealer has: 
[31m┌─────────┐[1B[11D│10       │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│       10│[1B[11D└─────────┘
[7A[12C┌─────────┐[1B[11D│░░░░░░░░░│[1B[11D│░ J I T ░│[1B[11D│░ T E R ░│[1B[11D│░ T E D ░│[1B[11D│░░░░░░░░░│[1B[11D└─────────┘
Player has: 
[30m┌─────────┐[1B[11D│J        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        J│[1B[11D└─────────┘[6A[1C[30m┌─────────┐[1B[11D│9        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        9│[1B[11D└─────────┘
 (19)
[H]it or [S]tand?
[2J[1;1HDealer has: 
[31m┌─────────┐[1B[11D│10       │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│       10│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│J        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        J│[1B[11D└─────────┘
 (20)

Player has: 
[30m┌─────────┐[1B[11D│J        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        J│[1B[11D└─────────┘[6A[1C[30m┌─────────┐[1B[11D│9        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        9│[1B[11D└─────────┘
 (19)
You lost to the Dealer. 💸
[m

[107m[2J[1;1H[32mWelcome to[31m Jitterted's[30m BlackJack
[2J[1;1HDealer has: 
[30m┌─────────┐[1B[11D│K        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        K│[1B[11D└─────────┘
[7A[12C┌─────────┐[1B[11D│░░░░░░░░░│[1B[11D│░ J I T ░│[1B[11D│░ T E R ░│[1B[11D│░ T E D ░│[1B[11D│░░░░░░░░░│[1B[11D└─────────┘
Player has: 
[31m┌─────────┐[1B[11D│2        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        2│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│K        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        K│[1B[11D└─────────┘
 (12)
[H]it or [S]tand?
[2J[1;1HDealer has: 
[30m┌─────────┐[1B[11D│K        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        K│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│10       │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│       10│[1B[11D└─────────┘
 (20)

Player has: 
[31m┌─────────┐[1B[11D│2        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        2│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│K        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        K│[1B[11D└─────────┘
 (12)
You lost to the Dealer. 💸
[m

[107m[2J[1;1H[32mWelcome to[31m Jitterted's[30m BlackJack
[2J[1;1HDealer has: 
[30m┌─────────┐[1B[11D│7        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        7│[1B[11D└─────────┘
[7A[12C┌─────────┐[1B[11D│░░░░░░░░░│[1B[11D│░ J I T ░│[1B[11D│░ T E R ░│[1B[11D│░ T E D ░│[1B[11D│░░░░░░░░░│[1B[11D└─────────┘
Player has: 
[31m┌─────────┐[1B[11D│4        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        4│[1B[11D└─────────┘[6A[1C[30m┌─────────┐[1B[11D│Q        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        Q│[1B[11D└─────────┘
 (14)
[H]it or [S]tand?
[2J[1;1HDealer has: 
[30m┌─────────┐[1B[11D│7        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        7│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│K        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        K│[1B[11D└─────────┘
 (17)

Player has: 
[31m┌─────────┐[1B[11D│4        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        4│[1B[11D└─────────┘[6A[1C[30m┌─────────┐[1B[11D│Q        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        Q│[1B[11D└─────────┘
 (14)
You lost to the Dealer. 💸
[m

[107m[2J[1;1H[32mWelcome to[31m Jitterted's[30m BlackJack
[2J[1;1HDealer has: 
[31m┌─────────┐[1B[11D│7        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        7│[1B[11D└─────────┘
[7A[12C┌─────────┐[1B[11D│░░░░░░░░░│[1B[11D│░ J I T ░│[1B[11D│░ T E R ░│[1B[11D│░ T E D ░│[1B[11D│░░░░░░░░░│[1B[11D└─────────┘
Player has: 
[31m┌─────────┐[1B[11D│A        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        A│[1B[11D└─────────┘[6A[1C[30m┌─────────┐[1B[11D│4        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        4│[1B[11D└─────────┘
 (15)
[H]it or [S]tand?
[2J[1;1HDealer has: 
[31m┌─────────┐[1B[11D│7        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        7│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│10       │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│       10│[1B[11D└─────────┘
 (17)

Player has: 
[31m┌─────────┐[1B[11D│A        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        A│[1B[11D└─────────┘[6A[1C[30m┌─────────┐[1B[11D│4        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        4│[1B[11D└─────────┘
 (15)
You lost to the Dealer. 💸
[m

[107m[2J[1;1H[32mWelcome to[31m Jitterted's[30m BlackJack
[2J[1;1HDealer has: 
[30m┌─────────┐[1B[11D│9        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        9│[1B[11D└─────────┘
[7A[12C┌─────────┐[1B[11D│░░░░░░░░░│[1B[11D│░ J I T ░│[1B[11D│░ T E R ░│[1B[11D│░ T E D ░│[1B[11D│░░░░░░░░░│[1B[11D└─────────┘
Player has: 
[31m┌─────────┐[1B[11D│5        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        5│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│3        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        3│[1B[11D└─────────┘
 (8)
[H]it or [S]tand?
[2J[1;1HDealer has: 
[30m┌─────────┐[1B[11D│9        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        9│[1B[11D└─────────┘[6A[1C[30m┌─────────┐[1B[11D│2        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        2│[1B[11D└─────────┘[6A[1C[30m┌─────────┐[1B[11D│J        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        J│[1B[11D└─────────┘
 (21)

Player has: 
[31m┌─────────┐[1B[11D│5        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        5│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│3        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        3│[1B[11D└─────────┘
 (8)
You lost to the Dealer. 💸
[m

[107m[2J[1;1H[32mWelcome to[31m Jitterted's[30m BlackJack
[2J[1;1HDealer has: 
[30m┌─────────┐[1B[11D│3        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        3│[1B[11D└─────────┘
[7A[12C┌─────────┐[1B[11D│░░░░░░░░░│[1B[11D│░ J I T ░│[1B[11D│░ T E R ░│[1B[11D│░ T E D ░│[1B[11D│░░░░░░░░░│[1B[11D└─────────┘
Player has: 
[30m┌─────────┐[1B[11D│8        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        8│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│3        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        3│[1B[11D└─────────┘
 (11)
[H]it or [S]tand?
[2J[1;1HDealer has: 
[30m┌─────────┐[1B[11D│3        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        3│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│4        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        4│[1B[11D└─────────┘[6A[1C[30m┌─────────┐[1B[11D│7        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        7│[1B[11D└─────────┘[6A[1C[30m┌─────────┐[1B[11D│A        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        A│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│2        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        2│[1B[11D└─────────┘
 (17)

Player has: 
[30m┌─────────┐[1B[11D│8        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        8│[1B[11D└─────────┘[6A[1C[31m┌─────────┐[1B[11D│3        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        3│[1B[11D└─────────┘
 (11)
You lost to the Dealer. 💸
[m

